### PR TITLE
Feature/nuget registration base

### PR DESCRIPTION
**Pull request by yamaoto**
Implement RegistrationsBaseUrl API #9

In public nuget.org feed RegistrationsBaseUrl does not match with documentation.
So I left as is for default listing API - SearchAutocompleteService
And for another feeds we can use RegistrationsBaseUrl by setting ListingApi option

### CODE REVIEW

File: Cynosura.Studio.Generator/PackageFeed/Models/RegistrationData.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+using System.Collections.Generic;
+
+namespace Cynosura.Studio.Generator.PackageFeed.Models
+{
+    public class RegistrationData
+    {
+        public   IEnumerable<RegistrationDataItem> Items { get; set; }
```
Comment (bacr) [inline Cynosura.Studio.Generator/PackageFeed/Models/RegistrationData.cs]: Unnecessary spaces

File: Cynosura.Studio.Generator/PackageFeed/NugetFeed.cs [hunk 2]
```diff
@@ -113,46 +132,48 @@ public class FeedData
...
-                    if (compare != 0)
-                        return compare;
                 }
-                return 0;
+                
+                var target = new NugetVersion(startVersion);
+                return versions
+                    .Where(w => w.Value.Version >= target.Version)
+                    .OrderByDescending(d => d.Value.Version)
+                    .Select(s=>s.Key)
```
Comment (bacr) [inline Cynosura.Studio.Generator/PackageFeed/NugetFeed.cs]: Missing spaces

File: Cynosura.Studio.Generator/PackageFeed/NugetFeed.cs [hunk 3]
```diff
@@ -39,37 +41,54 @@ private HttpClient GetHttpClient()
...
-            var searchAutocompleteService = await GetSearchAutocompleteServiceAsync();
-            var versionsUrl = $"{searchAutocompleteService}?id={packageName.ToLower()}&prerelease=true";
             var httpClient = GetHttpClient();
-            var versionsResult = await httpClient.GetStringAsync(versionsUrl);
-            var versions = versionsResult.DeserializeFromJson<VersionData>();
-            return OrderVersionsDescending(versions.Data);
+            if (_settings.ListingApi == NugetListingApi.RegistrationsBaseUrl)
+            {
+                var registrationsBase = await GetRegistrationsBaseUrlAsync();
+                var versionsUrl = $"{registrationsBase}/{packageName.ToLower()}";
```
Comment (bacr) [inline Cynosura.Studio.Generator/PackageFeed/NugetFeed.cs]: It must be $"{registrationsBase}{packageName.ToLower()}/index.json". Then it works for nuget.org too

Comment (bacr) [reply Cynosura.Studio.Generator/PackageFeed/NugetFeed.cs]: Also need to add
`var handler = new HttpClientHandler
            {
                AutomaticDecompression = DecompressionMethods.GZip
            };
            var httpClient = new HttpClient(handler);`

**Review by bacr — CHANGES_REQUESTED** [review]

File: Cynosura.Studio.Generator/PackageFeed/Models/RegistrationData.cs (hunk 1 above)
Comment (yamaoto) [reply Cynosura.Studio.Generator/PackageFeed/Models/RegistrationData.cs]: Resolved

File: Cynosura.Studio.Generator/PackageFeed/NugetFeed.cs (hunk 2 above)
Comment (yamaoto) [reply Cynosura.Studio.Generator/PackageFeed/NugetFeed.cs]: Resolved

File: Cynosura.Studio.Generator/PackageFeed/NugetFeed.cs (hunk 3 above)
Comment (yamaoto) [reply Cynosura.Studio.Generator/PackageFeed/NugetFeed.cs]: Resolved

**Review by bacr — APPROVED** [review]

**Pull request merged**